main.py

```diff
@@ -5,103 +5,135 @@
-    response = {
-        'error': {
-            'details': error.description,
-            'code': error.code
-        }
-    }
+    response = {"error": {"details": error.description, "code": error.code}}
 
     return jsonify(response), error.code
 
+
 def paginate_results(f):
     @functools.wraps(f)
     def decorated_function(*args, **kwargs):
-        limit = int(request.args.get('limit', 50))
-        page = int(request.args.get('page', 1))
+        limit = int(request.args.get("limit", 50))
+        page = int(request.args.get("page", 1))
 
-        queryset = f(*args, **kwargs).paginate(page=page, per_page=limit, max_per_page=100)
+        queryset = f(*args, **kwargs).paginate(
+            page=page, per_page=limit, max_per_page=100
+        )
         result = {
-            'data': [x.serialize() for x in queryset.items],
-            'total': queryset.total,
-            'limit': queryset.per_page,
-            'previous': queryset.prev_num,
-            'next': queryset.next_num
+            "data": [x.serialize() for x in queryset.items],
+            "total": queryset.total,
+            "limit": queryset.per_page,
+            "previous": queryset.prev_num,
+            "next": queryset.next_num,
         }
         return jsonify(result)
+
     return decorated_function
 
+
 def single_resource(f):
     @functools.wraps(f)
     def decorated_function(*args, **kwargs):
         result = f(*args, **kwargs).first_or_404()
         result = result.serialize()
         return jsonify(result)
+
     return decorated_function
 
-@app.route('/', methods=['GET'])
+
+@app.route("/", methods=["GET"])
 def home():
     return "<h1>Welcome to sunnah.com API.</p>"
 
-@app.route('/v1/collections', methods=['GET'])
+
+@app.route("/v1/collections", methods=["GET"])
 @paginate_results
 def api_collections():
     return HadithCollection.query.order_by(HadithCollection.collectionID)
 
-@app.route('/v1/collections/<string:name>', methods=['GET'])
+
+@app.route("/v1/collections/<string:name>", methods=["GET"])
 @single_resource
 def api_collection(name):
     return HadithCollection.query.filter_by(name=name)
 
-@app.route('/v1/collections/<string:name>/books', methods=['GET'])
+
+@app.route("/v1/collections/<string:name>/books", methods=["GET"])
 @paginate_results
 def api_collection_books(name):
     return Book.query.filter_by(collection=name).order_by(func.abs(Book.ourBookID))
 
-@app.route('/v1/collections/<string:name>/books/<string:bookNumber>', methods=['GET'])
+
+@app.route("/v1/collections/<string:name>/books/<string:bookNumber>", methods=["GET"])
 @single_resource
 def api_collection_book(name, bookNumber):
     book_id = Book.get_id_from_number(bookNumber)
     return Book.query.filter_by(collection=name, ourBookID=book_id)
 
-@app.route('/v1/collections/<string:collection_name>/books/<string:bookNumber>/hadiths', methods=['GET'])
+
+@app.route(
+    "/v1/collections/<string:collection_name>/books/<string:bookNumber>/hadiths",
```

Comment:
I'd suggest increasing the max line length instead of breaking into different lines in some `app.route` registrations and using single line in others? It would be better to have it consistent throughout the file.
 Also max line length is pretty controversial :) !

Reply:
Good idea, ill use Django's default 119

Reply:
a warning though that this might compress dictionary representations too especially in the `serialise()` methods of `models.py` @Glaisher

Reply:
> a warning though that this might compress dictionary representations too especially in the `serialise()` methods of `models.py` @Glaisher

Oh is that done by black autoformatting? I guess it's not much of an issue. :)